In [1]:
%load_ext autoreload
%autoreload 2

# Data Loading and Exploratory Data Analysis (EDA)

## Data Loading

In [2]:
# Importing libraries

import numpy as np
import pandas as pd

In [3]:
# Load the data
df = pd.read_csv('../data/raw/water_sanitation.csv')
df.head()

entry_id   country  \
0  81a69ab1-6070-4643-ab31-062dabbe355d     Haiti   
1  e4774c0e-38ca-48bc-a62d-56ad7dc2ebf1  Ethiopia   
2  ed103eee-2e27-4be4-b6fe-59e1b095a507     Nepal   
3  4d8e529d-08cd-428b-b4a8-e4c4cbadeb33     India   
4  8396148f-36a8-4ee9-af7f-653b8d06aa9c     Kenya   

                                        source_title  \
0                      Skill may certainly off sing.   
1                        Cup movie than fund artist.   
2  Note Congress day million everybody environmen...   
3                             Summer today southern.   
4                 Region sea nation fast shake born.   

                                             excerpt  geo_location  \
0  Health inspectors on site found that improper ...      5.937200   
1  The latest sanitation review indicated that im...     34.615215   
2  According to recent field observations, waste ...    -68.387244   
3  The latest sanitation review indicated that ef...    -71.814776   
4  A comprehensive report issued by the agency no...    -10.132914   

     water_source    sanitation_type                 sector  gender age_group  \
0           River    Open Defecation                 Health  Female     Youth   
1        Borehole  Composting Toilet  Public Infrastructure    Male     Youth   
2  Truck Delivery        Pit Latrine                 Health  Female     Adult   
3       Rainwater  Composting Toilet              Education    Male   Elderly   
4        Borehole  Composting Toilet  Public Infrastructure  Female     Youth   

   ... water_color water_clarity water_odor recent_rain  nearby_activities  \
0  ...       clear         clear   chemical          no            farming   
1  ...       murky        turbid     sewage         yes            farming   
2  ...       green         clear     sewage          no             mining   
3  ...       brown        turbid     sewage          no             mining   
4  ...       murky        cloudy       none          no               none   

  nearby_infrastructure vegetation_near_source waste_disposal_nearby  \
0      waste facilities                   lush         managed sites   
1      waste facilities                 sparse                  none   
2      waste facilities                   dead         managed sites   
3          sewage lines                   dead         managed sites   
4      waste facilities                   lush          open dumping   

  animal_access water_flow_rate  
0            no            slow  
1           yes        stagnant  
2           yes         flowing  
3           yes            slow  
4            no        stagnant  

[5 rows x 29 columns]

## Data Exploration

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4157 entries, 0 to 4156
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   entry_id                4157 non-null   object 
 1   country                 4157 non-null   object 
 2   source_title            4157 non-null   object 
 3   excerpt                 4157 non-null   object 
 4   geo_location            4157 non-null   float64
 5   water_source            4157 non-null   object 
 6   sanitation_type         4157 non-null   object 
 7   sector                  4157 non-null   object 
 8   gender                  4157 non-null   object 
 9   age_group               4157 non-null   object 
 10  income_level            4157 non-null   object 
 11  infrastructure_type     4157 non-null   object 
 12  population_density      4157 non-null   object 
 13  climate_zone            4157 non-null   object 
 14  year                    4157 non-null   

In [5]:
df.columns

Index(['entry_id', 'country', 'source_title', 'excerpt', 'geo_location',
       'water_source', 'sanitation_type', 'sector', 'gender', 'age_group',
       'income_level', 'infrastructure_type', 'population_density',
       'climate_zone', 'year', 'organization', 'source_type',
       'special_needs_group', 'severity', 'water_color', 'water_clarity',
       'water_odor', 'recent_rain', 'nearby_activities',
       'nearby_infrastructure', 'vegetation_near_source',
       'waste_disposal_nearby', 'animal_access', 'water_flow_rate'],
      dtype='object')

In [7]:
df['excerpt'][0]

'Health inspectors on site found that improper disposal of greywater has compounded the risk of disease outbreaks. Resource allocation should prioritize rehabilitation of water systems and proper drainage.'

From the above explorartion our data contains 29 columns and 4157 rows of entries.

Some of the columns include **'excerpt'** which gives an observation of the environment around a water source or water point or just a decscription of how the water looks like. 

Other columns includes '**water_odor'** which clasifies water smells into either none, chemical and sewage while **'water_color'** clasifies water color color into either clear, brown, murky or green

Most of this Columns describe how the condition of water and the sourounding environment which might affect or give an indication if the water is contaminated or not

## Data Cleaning

### Missing/Null-values

In [10]:
# Checking for missing/null values
for col in  df.columns:
    if df[col].isna().sum() > 0:
        print(f'{col} : {df[col].isna().sum()}')

special_needs_group : 779


- From the above cell only **'special_needs_group'** column contains 779 missing / null-values

In [11]:
df['special_needs_group'].value_counts()

special_needs_group
Children Under 5    891
Elderly             834
Disabled            831
Pregnant Women      822
Name: count, dtype: int64

In [13]:
# Replace missing with 'none' instead of having 'None' which reads as Null-Value
df['special_needs_group'] = df['special_needs_group'].fillna('none')
df['special_needs_group'].value_counts()

special_needs_group
Children Under 5    891
Elderly             834
Disabled            831
Pregnant Women      822
none                779
Name: count, dtype: int64

### Duplicates

In [16]:
# Checking for duplicated entries
df.duplicated().sum()

1157